carta/image.py

```diff
@@ -65,8 +67,12 @@ def new(cls, session, path, hdu, append):
         """
         path = session.resolve_file_path(path)
         directory, file_name = posixpath.split(path)
-        image_id = session.call_action("appendFile" if append else "openFile", directory, file_name, hdu, return_path="frameInfo.fileId")
-
+        if complex is not None:
+            file_name = f'{complex}("{file_name}")'
+            image_arithmetic = "true"
+            image_id = session.call_action("appendFile" if append else "openFile", directory, file_name, hdu, image_arithmetic, return_path="frameInfo.fileId")
+        else:
+            image_id = session.call_action("appendFile" if append else "openFile", directory, file_name, hdu, return_path="frameInfo.fileId")
```

Comment:
I would suggest doing a single call at the end outside the conditionals. Add an `image_arithmetic = False` either before the `complex is not None` conditional or in an `else`, and then you can do the call once at the end (including the `image_arithmetic` parameter).

If you like, maybe move the inline conditional for the action name from the call into its own line -- `command = "appendFile" if append else "openFile"`, and then use `command` in the call.